### PR TITLE
feat: API routes

### PR DESCRIPTION
**Pull request by rishabh3112**
Migrate code from `@rishabh3112/pc` and expose API routes for `/files` and `/chunks`.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "moduleResolution": "node",
     "resolveJsonModule": true,
     "isolatedModules": true,
+    "downlevelIteration": true,
```
Comment (chinmaychaudhary) [inline tsconfig.json]: why is this needed?

File: pages/api/files.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+type Data = {
+  directory: string;
+  pages: any;
+};
+
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse<Data>) {
+  const query: any = req.query.q || '';
+  const root = await fetch('http://localhost:3000/args/root').then((res) => res.text());
+  const files = await getAllFiles(root, query);
```
Comment (chinmaychaudhary) [inline pages/api/files.ts:12]: remove pagination for now.

File: tsconfig.json (hunk 1 above)
Comment (rishabh3112) [reply tsconfig.json]: To allow spread operation on `Set`, to better support iteration using new syntax `for of`, etc.
https://www.typescriptlang.org/tsconfig#downlevelIteration

File: pages/api/files.ts (hunk 2 above)
Comment (rishabh3112) [reply pages/api/files.ts:12]: Sure, will do.

Comment (rishabh3112) [reply pages/api/files.ts:12]: Done.

File: tsconfig.json (hunk 1 above)
Comment (rishabh3112) [reply tsconfig.json]: Updated target of typescript config by referencing to https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping

**Review by chinmaychaudhary — APPROVED** [review]
